lib/node/nodes/weighted-centroid.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'weighted-centroid';
+
+var PARAMS = {
+    source : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    weight_column : Node.PARAM.STRING,
+    category_column : Node.PARAM.STRING
+};
+
+var WeightedCentroid  = Node.create(TYPE, PARAMS, {cache: false});
+
+module.exports = WeightedCentroid;
+module.exports.TYPE = TYPE;
+module.exports.PARAMS = PARAMS;
+
+var weightedCentroidTemplate = dot.template([
+    'SELECT the_geom , class',
```

Comment:
I'm not fully sure, but I just realised that this might happen in other analyses as well. Actually the buffer/trade-area analyses don't return cartodb_id depending on some options.

So feel free to skip that for now. I will take care of the cartodb_id at a higher level in camshaft.